### ISSUE TITLE
Add specific example for using private Terraform modules to docs

### ISSUE THREAD

**Issue by ansgarm** [hashicorp/terraform-cdk#956]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

basically [this example](https://github.com/hashicorp/terraform-cdk/issues/951#issuecomment-913523817) plus a short description.

### References

* https://github.com/hashicorp/terraform-cdk/issues/951

**Comment by jsteinich** [comment]
#929 should make it so using the object form isn't required; however, having an example is the docs is still a good thing.

**Comment by xiehan** [comment]
Relates to #2072 - might make sense to tackle both at the same time.
